tasks/rh-sign-image/rh-sign-image.yaml

```diff
@@ -54,19 +93,79 @@ spec:
+        fi
+
+        REQUESTTYPE=$(jq -r '.sign.requestType // "internal-request"' "${DATA_FILE}")
+        service_account_name=$(jq -r '.spec.pipeline.serviceAccountName // "appstudio-pipeline"' "${RPA_FILE}")
+        if [ "${REQUESTTYPE}" == "internal-pipelinerun" ] ; then
+          requestType=internal-pipelinerun
+          serviceAccountName=${service_account_name}
+          taskGitUrl=$(params.taskGitUrl)
+          taskGitRevision=$(params.taskGitRevision)
+          defaultRequest="simple-signing-pipeline"
+        else
+          requestType=internal-request
+          # we do not need to set these for internal-requests
+          serviceAccountName=
+          taskGitUrl=
+          taskGitRevision=
+          defaultRequest="hacbs-signing-pipeline"
+        fi
+        # shellcheck disable=SC2086
+        request=$(jq --arg defaultRequest "${defaultRequest}" -r '.sign.request // $defaultRequest' ${DATA_FILE})
+
+        serviceAccountArg=
+        if [ -n "${serviceAccountName}" ]; then
+          serviceAccountArg="--service-account ${serviceAccountName}"
+        fi
+        taskGitUrlArg=
+        if [ -n "${taskGitUrl}" ]; then
+          taskGitUrlArg="--task-git-url ${taskGitUrl}"
+        fi
+        taskGitRevisionArg=
+        if [ -n "${taskGitRevision}" ]; then
+          taskGitRevisionArg="--task-git-revision ${taskGitRevision}"
+        fi
 
         # make sure RUNDIR exists
         mkdir -p $RUNDIR
 
         default_pipeline_image="quay.io/redhat-isv/operator-pipelines-images:released"
-        request=$(jq -r '.sign.request // "hacbs-signing-pipeline"' ${DATA_FILE})
+
         pipeline_image=$(jq -r --arg default_pipeline_image ${default_pipeline_image} \
             '.sign.pipelineImage // $default_pipeline_image' ${DATA_FILE})
         config_map_name=$(jq -r '.sign.configMapName // "signing-config-map"' ${DATA_FILE})
         defaultPushSourceContainer=$(jq -r '.mapping.defaults.pushSourceContainer' ${DATA_FILE})
 
-        N=$(params.concurrentLimit)  # The maximum number of images to be processed at once
-        count=0
+        if [[ "$(params.pyxisServer)" == "production" ]]
+        then
+          PYXIS_GRAPHQL_URL="https://graphql-pyxis.api.redhat.com/graphql/"
+        elif [[ "$(params.pyxisServer)" == "stage" ]]
+        then
+          PYXIS_GRAPHQL_URL="https://graphql-pyxis.preprod.api.redhat.com/graphql/"
+        elif [[ "$(params.pyxisServer)" == "production-internal" ]]
+        then
+          PYXIS_GRAPHQL_URL="https://graphql.pyxis.engineering.redhat.com/graphql/"
+        elif [[ "$(params.pyxisServer)" == "stage-internal" ]]
+        then
+          PYXIS_GRAPHQL_URL="https://graphql.pyxis.stage.engineering.redhat.com/graphql/"
+        else
+          echo "Invalid pyxisServer parameter. Only 'production','production-internal,'stage-internal' \
+              and 'stage' allowed."
+          exit 1
+        fi
+        set +x
+        echo "${pyxisCert:?}" > /tmp/crt
```

Comment:
Why the `:?`? These variables are mandatory, right? It won't work without them. Did shellcheck complain about this? I would be surprised, since we have it in several places, but maybe we just didn't run shellcheck there yet?

Reply:
Yes. shellcheck complain about this....and we have the same pattern of `echo "${pyxisCert:?}" > /tmp/crt` in tasks like `create-pyxis-image` but I believe they have not need **shellchecked** yet.

Reply:
OK, fine with me then.